kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -506,6 +506,90 @@ class SparseMatrixMultiplicationUtility
         delete[] aux_val_new_a;
     }
 
+    /**
+     * @brief This method computes of the transpose matrix of a given matrix
+     * @param rA The resulting matrix
+     * @param rB The second matrix to transpose
+     */
+    template <class AMatrix, class BMatrix>
+    static void TransposeMatrix(
+        AMatrix& rA,
+        const BMatrix& rB,
+        const double Factor = 1.0
+        )
+    {
+        // Get access to B data
+        const std::size_t* index1 = rB.index1_data().begin();
+        const std::size_t* index2 = rB.index2_data().begin();
+        const double* data = rB.value_data().begin();
+        const std::size_t transpose_nonzero_values = rB.value_data().end() - rB.value_data().begin();
+
+        const std::size_t size_system_1 = rB.size1();
+        const std::size_t size_system_2 = rB.size2();
+
+        if (rA.size1() != size_system_2 || rA.size2() != size_system_1 ) {
+            rA.resize(size_system_2, size_system_1);
+        }
+
+        std::size_t* new_a_ptr = new std::size_t[size_system_2 + 1];
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(size_system_2 + 1); ++i)
+            new_a_ptr[i] = 0;
+
+        std::ptrdiff_t* aux_index2_new_a = new std::ptrdiff_t[transpose_nonzero_values];
+        double* aux_val_new_a = new double[transpose_nonzero_values];
+
+        #pragma omp parallel for
+        for (int i=0; i<static_cast<int>(size_system_1); ++i) {
+            std::size_t row_begin = index1[i];
+            std::size_t row_end   = index1[i+1];
+
+            for (std::size_t j=row_begin; j<row_end; j++) {
+                #pragma omp atomic
+                new_a_ptr[index2[j] + 1] += 1;
+            }
+        }
+
+        // We initialize the blocks sparse matrix
+        std::partial_sum(new_a_ptr, new_a_ptr + size_system_2 + 1, new_a_ptr);
+
+        std::size_t* aux_indexes = new std::size_t[size_system_2];
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(size_system_2); ++i)
+            aux_indexes[i] = 0;
+
+//         #pragma omp parallel for
```

Comment:
@roigcarlo any idea how to activate thsi loop in parallel?

Reply:
i don't think it is a good idea to nest two parallel loops. It IS supported in openmp, however it is still not a good idea to do that

Reply:
There are not nested, the first one is just one loop to fill the vector with zeros

Reply:
oh it is a case of "misleading indent"

Reply:
One sec